tls/s2n_key_log.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <s2n.h>
+#include "tls/s2n_config.h"
+#include "tls/s2n_connection.h"
+#include "tls/s2n_crypto_constants.h"
+#include "tls/s2n_quic_support.h" /* this currently holds the s2n_secret_type_t enum */
```

Comment:
Should we put the enum somewhere else?

Reply:
Probably should but it seems tangential to this PR

Reply:
If this is the first feature other than QUIC that uses that enum, putting that enum where non-QUIC features can access it (without commenting on their strange imports) seems pretty relevant to this PR.

Reply:
Do you have an idea of where you want it moved?

Reply:
One thing to keep in mind when moving it is it needs to be somewhat publicly consumable because that quic_support.h will be pulled in by quic implementations so it's probably a good idea not to pull in a bunch of other private stuff.

Reply:
That's a good point. We can't put it in most internal headers bc we'd pull in other private stuff. Even if we put it in its own new private header, it wouldn't be clear why that header existed and it would probably collect non-public cruft.

I guess leaving it in quic_support.h with a comment is probably the best option for now.